### PR TITLE
Add GitHub action to publish to PyPI

### PR DESCRIPTION
**Pull request by zhuchcn**
## Description

Closes #...  

## Checklist

- [X] This PR does NOT contain [PHI](https://ohrpp.research.ucla.edu/hipaa/) or germline genetic data.  A repo may need to be deleted if such data is uploaded. Disclosing PHI is a [major problem](https://healthitsecurity.com/news/ucla-health-reaches-7.5m-settlement-over-2015-breach-of-4.5m).
- [X] This PR does NOT contain molecular files, compressed files, output files such as images (*e.g.* `.png`, .`jpeg`), `.pdf`, `.RData`, `.xlsx`, `.doc`, `.ppt`, or other non-plain-text files.  To automatically exclude such files using a [.gitignore](https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files) file, see [here](https://github.com/uclahs-cds/template-base/blob/main/.gitignore) for example.
- [X] I have read the [code review guidelines](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3187646/Code+Review+Guidelines) and the [code review best practice on GitHub check-list](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3189956/Code+Review+Best+Practice+on+GitHub+-+Check+List).
- [X] The name of the branch is meaningful and well formatted following the [standards](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3189956/Code+Review+Best+Practice+on+GitHub+-+Check+List), using [AD_username (or 5 letters of AD if AD is too long)]-[brief_description_of_branch].
- [X] I have added the major changes included in this pull request to the `CHANGELOG.md` under the next release version or unreleased, and updated the date.
- [X] All test cases passed locally.

### CODE REVIEW

**Comment by zhuchcn** [comment]
@aholmes I tested it locally to push it to test-PyPI and it does work (https://test.pypi.org/project/mopepgen/). Installed it from test-PyPI and verified that the "test" directory isn't installed as a module. This workflow runs on tags, and it recognizes the version from `setup.cfg` directly. How should we test the action on test-PyPI? If we make a tag on the latest commit, it will still be published to test-PyPI as the current version number v1.3.2. Also, if I push a tag on the latest commit on this branch, will the action be triggered even when the main branch doesn't have the action yet?

**Comment by zhuchcn** [comment]
Seems like once the PyPI project is already created, I can't add a github action to it. The 'Add a new pending publisher' only works for project not exist on PyPI yet. Seems like we can provide a PyPI API token to the action. But for the actual PyPI, we should be able to use the 'Add a new pending publisher' function directly without token.

File: .github/workflows/pypi.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+jobs:
+  build:
+    name: Build distribution
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout main
+        uses: actions/checkout@v2
+
+      - name: Build distribution
+        uses: uclahs-cds/tool-pypi-publish-action/build-distribution@a407d50
```
Comment (zhuchcn) [inline .github/workflows/pypi.yaml:18]: Does it work by adding commit ID after the subdirectory of the repo?

**Comment by aholmes** [comment]
> How should we test the action on test-PyPI?  If we make a tag on the latest commit, it will still be published to test-PyPI as the current version number v1.3.2.

Take a look at [this PR](https://github.com/uclahs-cds/BL_Python/pull/29) to get a sense of the steps I took in making this work for BL_Python.

During my final testing on test.pypi, I committed fake version bumps (to `main` it looks like, though I'm not sure that was necessary) and then pushed fake tags. For example, here I altered the version of `BL_Python.web` to [`0.0.1-16`](https://github.com/uclahs-cds/BL_Python/commit/8d0a6cf2753a45dc6d0ee550e56d02c75dd95221). I later [reverted](https://github.com/uclahs-cds/BL_Python/pull/29/commits/1c8f0dbb1a799672deb2b3cb6bebe3d169bbfe6e) this when my testing was done.

You can see all my test version bumps published on [test.pypi](https://test.pypi.org/project/BL-Python.web/#history). I don't recall exactly where the "post" part comes from - it might be automatic on PyPI's part?


> Also, if I push a tag on the latest commit on this branch, will the action be triggered even when the main branch doesn't have the action yet?

No - actions are only triggered if the workflow file exists in the default branch. However, that workflow file _can be_ empty and do nothing; it only has to exist. Once that's done, any branch should be able to trigger workflows from that file in that branch.

**Comment by aholmes** [comment]
> Seems like once the PyPI project is already created, I can't add a github action to it. The 'Add a new pending publisher' only works for project not exist on PyPI yet. Seems like we can provide a PyPI API token to the action. But for the actual PyPI, we should be able to use the 'Add a new pending publisher' function directly without token.

What steps did you take, and what failed where and when?

File: .github/workflows/pypi.yaml (hunk 1 above)
Comment (aholmes) [reply .github/workflows/pypi.yaml:18]: I think so, but you might need the full commit SHA.

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsuses

They don't show specifically using a "public action in a subdirectory" with a commit SHA, but they do show, separately, both `actions/checkout@8f4b7f84864484a7bf31766abe9204da3cbe65b3` and `actions/aws/ec2@main`.

Comment (zhuchcn) [reply .github/workflows/pypi.yaml]: Cool! Just updated

**Comment by zhuchcn** [comment]
So seems like we can first merge this to main, and then I can change the version to a random number, and push a tag to the branch to trigger the workflow? I won't commit version change to main so we don't have to revert.

I think I just successfully connected this repo to the project in test pypi so we can give it a try now.

**Comment by aholmes** [comment]
> So seems like we can first merge this to main, and then I can change the version to a random number, and push a tag to the branch to trigger the workflow?

Yep, that should do it.

**Review by aholmes — APPROVED** [review]

**Comment by zhuchcn** [comment]
[v0.0.2](https://test.pypi.org/manage/project/mopepgen/release/0.0.2/) successfully pushed to test.pypi!

**Pull request merged**